### DISCUSSION TITLE
Button Danger

### DISCUSSION THREAD

**Discussion by yceballost** [Telefonica/mistica-design#501] [Proposals]
Actualmente, el estilo de un button danger es como el Primary pero en color rojo. IMHO, tiene sentido que no exista un buttonSecondaryDanger porque se pensó como un botón de prioridad máxima para acciones la cual un usuario no podría volver a recuperar.

Pero! Creo que pueden existir acciones críticas de menor nivel. No es lo mismo borrar tu cuenta, que eliminar un ticket de soporte.

Tendría sentido tener ese buttonDanger para acciones destructivas críticas como el caso de la cuenta y tener un ButtonLinkDanger para casos destructivos pero no tan crítico? Siento la ambigüedad 🤣

![image](https://user-images.githubusercontent.com/6722153/175573930-d6d270e3-c9ff-449a-a076-5254691054fc.png)![image](https://user-images.githubusercontent.com/6722153/175573825-8ab0d627-6833-4864-a150-343d456e60bb.png)

**Comment by aweell** [comment]
Veo los casos de danger como una dimensión separada de las variantes jerárquicas del botón (primary, secondary) y me encaja totalmente lo que planteas, no siempre quiere dársele el mismo énfasis a una acción destructiva y en ocasiones podrá ser una acción de menor importancia dentro de esa escala.

<img width="873" alt="Screenshot 2022-06-24 at 19 05 27" src="https://user-images.githubusercontent.com/44420072/175608962-e077b59e-6804-40de-b0b9-9b9110f00a29.png">

También incluiría en esa dimensión el icon button abriendo la posibilidad a que existan acciones destructivas en esa variante.


El caso del button link lo veo un poco diferente por la connotación que tiene a nivel de interacción, y su relación directa a nivel de color con los anchor, quizás en este caso no debería existir una modalidad de danger? Quizás me choca por la propia nomenclatura, y si se llamara tertiary no vería esta problemática? (Puede ser...).

En cuanto al énfasis y su escala, veo que la modalidad danger siempre se va a colocar por delante, me imagino este cambio como algo así:


Emphasis | Button
-- | --
Yell !!!! | `Primary danger Button`
Shout !! | `Primary Button`, `Secondary danger button`
Cheer! | `Secondary Button`, `Icon danger button`
Murmur | `Icon Button`, `Link danger???`
Whistle | `Link Button` & `Contextual Button`

**Reply by yceballost** [reply]
Entiendo lo de la connotación, y quizás nunca debimos tener un ButtonLink como tal, y si que ese fuese un Tertiary sin que el visual fuese el que le diera el nombre.

Añado un poco más de contexto sobre dónde sale esto
Esto es una pantalla de Movistar en México
![image](https://user-images.githubusercontent.com/6722153/175615706-fb138cf4-c37b-4235-be0c-52f662b10a11.png)
Es bastante evidente que el buttonDanger en ese caso no funcionaría

y esto lo que le planteé
![image](https://user-images.githubusercontent.com/6722153/175615602-bc6f94d4-f0ed-462c-b773-d6b2d05b1c3a.png)

Igual estoy viciado con este ejemplo que propuse, pero creo que lo que propones me seguiría chocando a nivel de jerarquía, veo demasiada pelea en la convivencia con un primary, de ahí a que "me saltara" el secondary y directamente me fuese al buttonLink
![image](https://user-images.githubusercontent.com/6722153/175616677-0c98495b-42f8-45d3-857e-0497272ca0d6.png)

Tengo dudas de lo importante que pueda ser el nombre de ButtonLink como para que eso fuese un impedimento a la existencia de ese buttonLinkDanger (aunque te entiendo y comparto parte de tu duda)

![image](https://user-images.githubusercontent.com/6722153/175617046-4a23f203-bd8d-440f-8542-81d112e75d49.png)

**Comment by aweell** [comment]
Update: hasta ahora se incluía la definición del button link danger en la documentación de botones de BF. Se elimina en https://github.com/Telefonica/mistica-design/issues/599.

**Reply by MontseTelefonica** [reply]
No me eliminéis el buttonLink danger! que para el CMS me hace falta :)

**Reply by aweell** [reply]
No es que lo hayamos eliminado, es que en brandFactory aparecía pero no es posible a día de hoy utilizar un link danger. Podrías incluir ejemplos de los casos en los que necesitarías su uso en el CMS para añadir otro punto de vista al debate?

**Reply by MontseTelefonica** [reply]
En el caso del CMS, trabajamos mucho con tablas y listas, y ahí nos viene bien tener textLink dangers para acciones destructivas, véase ejemplo. 
<img width="383" alt="image" src="https://user-images.githubusercontent.com/95479574/231970107-ac7a774d-27eb-4d44-9411-b04c335ab7fe.png">

**Reply by AnaMontes11** [reply]
We've finally decided that it makes sense to have a buttonLinkDanger, just as you mentioned @MontseTelefonica but for now, we're holding off on including a buttonSecondaryDanger
